enterprise/server/remote_execution/copy_on_write/copy_on_write.go

```diff
@@ -686,40 +693,56 @@ func (m *Mmap) initMap() error {
 }
 
 func (m *Mmap) ReadAt(p []byte, off int64) (n int, err error) {
-	if !m.mapped {
+	if !m.safeReadMapped() {
 		if err := m.initMap(); err != nil {
 			return 0, err
 		}
 	}
-	if err := checkBounds("read", int64(len(m.data)), p, off); err != nil {
+	dataLen, err := m.SizeBytes()
+	if err != nil {
+		return 0, err
+	}
```

Comment:
if we move the bounds check after `RLock` / `RUnlock` then I think we can go back to using `len(m.data)` directly, which seems a bit simpler and doesn't result in the extra locking overhead incurred by the SizeBytes() call. Same in WriteAt below.